sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -95,7 +95,11 @@ case object REPARTITION_BY_NUM extends ShuffleOrigin
 // Indicates that the shuffle operator was added by the user-specified repartition operator. Spark
 // firstly tries to coalesce partitions, if it cannot be coalesced, then use the local shuffle
 // reader.
-case object REPARTITION_BY_NONE extends ShuffleOrigin
+case object REBALANCE_PARTITIONS_BY_NONE extends ShuffleOrigin
+
+// Indicates that the shuffle operator was not guaranteed the output partitioning so Spark
+// can try to optimize the partition number in AQE framework.
```

Comment:
please highlight the difference between this one and `REBALANCE_PARTITIONS_BY_NONE `